event-sourcing/subscription-manager/src/main/java/uk/gov/justice/services/event/sourcing/subscription/EventSourceProducer.java

```diff
@@ -0,0 +1,33 @@
+package uk.gov.justice.services.event.sourcing.subscription;
+
+import static uk.gov.justice.services.core.cdi.EventSourceName.DEFAULT_EVENT_SOURCE_NAME;
+
+import uk.gov.justice.services.eventsourcing.source.core.DefaultEventSource;
+import uk.gov.justice.services.eventsourcing.source.core.EventSource;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.enterprise.inject.Produces;
+import javax.enterprise.inject.spi.InjectionPoint;
+import javax.inject.Inject;
+
+@ApplicationScoped
+public class EventSourceProducer {
+
+    @Inject
+    EventSourceNameExtractor eventSourceNameExtractor;
+
+    @Inject
+    CdiEventSourceFactory cdiEventSourceFactory;
+
+    @Produces
+    public EventSource eventSource(final InjectionPoint injectionPoint) {
+
+        final String eventSourceName = eventSourceNameExtractor.getEventSourceNameFromQualifier(injectionPoint);
+
+        if (DEFAULT_EVENT_SOURCE_NAME.equals(eventSourceName)) {
+            return cdiEventSourceFactory.create(DefaultEventSource.class);
+        }
+
+        throw new EventSourceProducerException("Use of non default EventSources not yet implemented");
```

Comment:
This exception name is generic; it doesn't indicate the cause of the problem. Maybe use a CDI exception such as UnsatisfiedResolutionException?